build.zig

```diff
@@ -23,14 +23,31 @@ pub fn build(b: *std.build.Builder) void {
     const test_step = b.step("test", "Run library tests");
     test_step.dependOn(&main_tests.step);
 
+    inline for ([_][]const u8{ "triangle", "boids" }) |name| {
+        const example = b.addExecutable("example-" ++ name, "examples/" ++ name ++ "/main.zig");
+        example.setTarget(target);
+        example.setBuildMode(mode);
+        example.addPackage(pkg);
+        example.addPackage(gpu.pkg);
+        example.addPackage(glfw.pkg);
+        link(b, example, options);
+        example.install();
+
+        const example_run_cmd = example.run();
+        example_run_cmd.step.dependOn(b.getInstallStep());
+        const example_run_step = b.step("run-example-" ++ name, "Run the example");
+        example_run_step.dependOn(&example_run_cmd.step);
+    }
 
-    inline for ([_][] const u8{"triangle", "boids"}) |name| {
+    const zmath_pkg = @import("zmath/build.zig").pkg;
+    inline for ([_][]const u8{ "mandelbrot", "fragment-shader", "rotating-cube", "two-cubes" }) |name| {
```

Comment:
Would be better if both the example collection can be combined, maybe with an additional needs_zmath field or with a []Pkg additional_deps field. This would require switching to use a struct to represent the examples

Reply:
Seems like the best approach, you are right